### DISCUSSION TITLE
C++ and dealing with functions and constructors defined in the class

### DISCUSSION THREAD

**Discussion by Perpixel** [rust-lang/rust-bindgen#2617] [Q&A]
I've been reading tons of documentations all over but just can't find a definitive answer to dealing with this situation beside writing a c wrapper for everything defined inside a class header. ex;
```cpp
class MyClass
{
public:
    Myclass() { /* init something something */ }
};
```

generate-inline-function does create the binding but fails at link time, and anyway, I'm using a large proprietary static lib...  Anything like --wrap-static-fns-* is available?

**Comment by pvdrz** [comment]
unless you can recompile the static lib so inline items are exposed anyway there's no solution for this right now. `--wrap-static-fns` doesn't deal with c++ code yet

**Reply by pvdrz** [reply]
You need to be able to compile the static library against the C code generated by `--wrap-static-fns` so I'm not even sure if it is even possible to do this.
